pkg/provisioner/ironic/raid.go

```diff
@@ -210,10 +210,14 @@ func BuildRAIDCleanSteps(raidInterface string, target *metal3v1alpha1.RAIDConfig
 	}
 
 	// Hardware RAID
-	// Ignore SoftwareRAIDVolumes
-	if target != nil {
-		target.SoftwareRAIDVolumes = nil
+	// If hardware RAID configuration is nil,
+	// keep old hardware RAID configuration
+	if target == nil || target.HardwareRAIDVolumes == nil {
+		return
 	}
+
```

Comment:
@andfasano 
The missing that I think of is that:
In the current master implementation, if the target node does not have a RAID controller, the ironic node will enter the clean failed state because the delete_configuration clean step failed.

Therefore, our goal is:
- If hardwareRAIDVolume = nil
Keep actual RAID configuration
- If hardwareRAIDVolume = []
Clear actual RAID configuration
- Other
Apply RAID configuration to node

Reply:
Additional explaintion:

This is the missing case:
User apply a new bmh without `raid` field, and at the same time, the target node doesn't have RAID controller.

In above case, when entering BuildRAIDCleanSteps(), the values of target and actual are:
```go
target = {
    hardwareRAIDVolume = nil
    softwareRAIDVolume = []
}

actual = nil
```

Because of above values of target and actual, BMO will append `delete_configuration` into clean steps, then due to there is no RAID controller, an error will happen, which leads to reentry the preparing status (infinity loop).

Reply:
(I guess it's `This is because when a node does not have a RAID controller`_`but supports RAID`_)

Thanks @Hellcatlk for the clarification and links (and also for having used a different branch), it's definitely clearer now. Could you please update the commit message to reflect this case (ie that hw RAID operations are skipped when a node does not have a RAID controller), since right now it still sound pretty generic?

Reply:
Thanks, done.